scripts/blocks-utils.js

```diff
@@ -60,8 +62,44 @@ const formatDateTime = (date) => date && date.toLocaleString('en-US', {
   hour12: true,
 });
 
+function createPictureElement(
+  src,
+  alt = '',
+  eager = false,
+  breakpoints = [{ media: '(min-width: 600px)', width: '2000' }, { width: '750' }],
+) {
+  if (src.indexOf('http://') === -1 && src.indexOf('https://') === -1) {
+    return createOptimizedPicture(src, alt, eager, breakpoints);
+  }
+  const picture = document.createElement('picture');
+  const image = document.createElement('img');
+  image.setAttribute('src', src);
+  image.setAttribute('alt', alt);
+  image.setAttribute('loading', eager ? 'eager' : 'lazy');
+  picture.appendChild(image);
+  return picture;
+}
+
+function listenToScroll(elementToObserve, callback, threshold = 0.1) {
+  const observer = new IntersectionObserver((entries, observerInstance) => {
+    entries.forEach((entry) => {
+      if (entry.isIntersecting) {
+        callback(elementToObserve);
```

Comment:
I think we should consider adding a debounce or any other mechanism here to avoid too many callbacks (API calls/ animations) getting fired at once. For example consider navigating to the bottom most section by clicking a quick link, it would lead to all callbacks getting fired at once for elements intersecting and going away quickly.
cc @sdmcraft

Reply:
@vivesing  we might have to do this based on the performance when all blogs articles are in place and api responses are ready. I have opened a follow up issue https://github.com/aemsites/icicidirect/issues/65 for the same